### PR TITLE
Manage dependencies with bower and grunt

### PR DESCRIPTION
**Pull request by dj**
Just putting this up here for now, want to do a bit more before I merge

@benjaminleichter @alexjohnson505 

The new way to add JS libs (you may have to run `npm install` and `bower install` first):
1. Check if it's available from bower.io. If not, we can have a discussion about what to do about vendoring libraries.
2. Install with `bower install <package_name>`
3. Add name of the JS file to `Gruntfile.js`, [see here](https://github.com/data-visualization-capstone/web/compare/move_deps_to_bower?expand=1#diff-35b4a816e0441e6a375cd925af50752cR21)
4. run `grunt`

TODO:
- make grunt watch run concat
- uglify js files (I don't want to do this without figuring out source maps, so maybe I can do this later)
- source maps

### CODE REVIEW

**Comment by alexjohnson505** [comment]
Love it. I'm a huge fan of Bower & Grunt.

Todo:
- get moment.js, d3, underscore, and mapbox via bower
- add support for a "grunt watch" to compile .less files. See: http://jonathanmh.com/make-grunt-watch-for-lesscss-changes/

**Comment by dj** [comment]
@alexjohnson505 
wow, I just realized those are missing. I totally did this at some point but I accidentally lost it to the void (git)

**Comment by alexjohnson505** [comment]
Ha ha. I figured so. 

Let me know when you think this branch is good to merge! I made a few changes to 'development', so a merge conflict might happen.

**Comment by danabucci** [comment]
@dj let me know if I have to do anything on the Dev server to load the bower and grunt stuff. Grunt is installed globally already, but idk if anything needs to be started, stopped, reset, what have you.

**Comment by dj** [comment]
oh I see what I did. I forgot to use `bower install` with --save, and `bower_components` wasn't in VC.

I'm working finishing this up right now, I'll ping you to check it when it's done

**Comment by alexjohnson505** [comment]
Can you remove the compiled app.js file from version control?

File: css/style.css [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/* style.less
```
Comment (alexjohnson505) [inline css/style.css:1]: Lets ignore this file (style.css) from version control.

**Comment by alexjohnson505** [comment]
Couple minor notes:
- Ignore compiled files from version control (app.js / *.css).
- Backmerge branch.

**Pull request merged**